aerospike-mixin/alerts/alerts.libsonnet

```diff
@@ -0,0 +1,154 @@
+                ) % $._config,
+            },
+          },
+          {
+            alert: 'AerospikeUnavailablePartitions',
+            expr: |||
+              sum without(service) (aerospike_namespace_unavailable_partitions) > %(alertsCriticalUnavailablePartitions)s
+            ||| % $._config,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'There is an unavailable partition in the Aerospike cluster.',
+              description:
+                (
+                  '{{ printf "%%.0f" $value }} unavailable partition(s) in namespace {{$labels.ns}}, on node {{$labels.instance}}, on cluster {{$labels.aerospike_cluster}}, ' +
+                  'which is above the threshold of %(alertsCriticalUnavailablePartitions)s.'
+                ) % $._config,
+            },
+          },
+          {
+            alert: 'AerospikeDeadPartitions',
+            expr: |||
+              sum without(service) (aerospike_namespace_dead_partitions) > %(alertsCriticalDeadPartitions)s
+            ||| % $._config,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'There is a dead partition in the Aerospike cluster.',
+              description:
+                (
+                  '{{ printf "%%.0f" $value }} dead partition(s) in namespace {{$labels.ns}}, on node {{$labels.instance}}, on cluster {{$labels.aerospike_cluster}}, ' +
+                  'which is above the threshold of %(alertsCriticalDeadPartitions)s.'
+                ) % $._config,
+            },
+          },
+          {
+            alert: 'AerospikeNamespaceRejectingWrites',
+            expr: |||
+              sum without(service) (aerospike_namespace_stop_writes + aerospike_namespace_clock_skew_stop_writes) > %(alertsCriticalSystemRejectingWrites)s
+            ||| % $._config,
+            'for': '1m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'A namespace is currently rejecting all writes. Check for unavailable/dead partitions, clock skew, or nodes running out of memory/disk.',
+              description:
+                (
+                  'Namespace {{$labels.ns}} on node {{$labels.instance}} on cluster {{$labels.aerospike_cluster}} is currently rejecting all client-originated writes.'
+                ) % $._config,
+            },
+          },
+          {
+            alert: 'AerospikeHighClientReadErrorRate',
+            expr: |||
+              sum without(service) (rate(aerospike_namespace_client_read_error[5m])) / (clamp_min(sum without(service) (rate(aerospike_namespace_client_read_error[5m])) + sum without(service) (rate(aerospike_namespace_client_read_success[5m])), 1)) > %(alertsWarningHighClientReadErrorRate)s
```

Comment:
Nothing wrong with this, but this also should have worked just fine if you took out all of the "rates" (although it would mean something slightly different than with the rates)

Reply:
I don't think this alert is nearly as valuable without the rates